gusto/linear_solvers.py

```diff
@@ -72,9 +68,29 @@ class CompressibleSolver(TimesteppingSolver):
     :arg params (optional): solver parameters
     """
 
-    def __init__(self, state, quadrature_degree=None, params=None):
-
-        self.state = state
+    default_solver_parameters = {
+        'pc_type': 'fieldsplit',
+        'pc_fieldsplit_type': 'schur',
+        'ksp_type': 'gmres',
+        'ksp_max_it': 100,
+        'ksp_gmres_restart': 50,
+        'pc_fieldsplit_schur_fact_type': 'FULL',
+        'pc_fieldsplit_schur_precondition': 'selfp',
+        'fieldsplit_0_ksp_type': 'preonly',
+        'fieldsplit_0_pc_type': 'bjacobi',
+        'fieldsplit_0_sub_pc_type': 'ilu',
+        'fieldsplit_1_ksp_type': 'preonly',
+        'fieldsplit_1_pc_type': 'gamg',
+        'fieldsplit_1_mg_levels_ksp_type': 'chebyshev',
+        'fieldsplit_1_mg_levels_ksp_chebyshev_esteig': True,
+        'fieldsplit_1_mg_levels_ksp_chebyshev_esteig_random': True,
+        'fieldsplit_1_mg_levels_ksp_max_it': 1,
+        'fieldsplit_1_mg_levels_pc_type': 'bjacobi',
+        'fieldsplit_1_mg_levels_sub_pc_type': 'ilu'
+    }
+
+    def __init__(self, state, quadrature_degree=None, solver_parameters=None,
+                 overwrite_solver_parameters=False):
```

Comment:
I'll grab the options from your PR if that's OK - shallow water examples are currently broken on this branch because I haven't set any default solver options and it'd be silly to set old ones that I'm joust going to replace...